students/ArchWu/Session2/series.py

```diff
@@ -0,0 +1,47 @@
+def fib(n):
+    """Return nth fibonacci number with recursion."""
+    if n == 0:
+        return 0
+    if n == 1:
+        return 1
+    return fib(n - 1) + fib (n - 2)
+
+def fib2(n):
+    """Another function that returns nth fibonacci number with lower time and space used."""
+    a, b = 0, 1
+    if n == 0:
+        return 0
+    for _ in range(n - 1):
+        a, b = b, a + b
+    return b
+
+def lucas(n):
+    """Return nth lucas number with low time and space complexity: O(n) time complexity and O(1) space complexity."""
+    a, b = 2, 1
+    if n == 0:
+        return 2
+    for _ in range(n - 1):
+        a, b = b, a + b
+    return b
+
+
+def test_fib():
```

Comment:
Don't forget the sum_series, refer to the Generalizing area: https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/fib_and_lucas.html

Reply:
I will add them shortly!